src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs

```diff
@@ -280,5 +294,36 @@ private static string UpdateExtensionsFolderInRunSettings(string settingsXml)
                 return document.OuterXml;
             }
         }
+
+        /// <summary>
+        /// Collect Enabled Data Collectors
+        /// </summary>
+        private void CollectEnabledDataCollectors()
+        {
+            var dataCollectionSettings = XmlRunSettingsUtilities.GetDataCollectionRunSettings(this.settingsXml);
+
+            if (dataCollectionSettings == null || !dataCollectionSettings.IsCollectionEnabled)
+            {
+                return;
+            }
+
+            var enabledDataCollectors = new List<DataCollectorSettings>();
```

Comment:
Do we need to create a new list?

Reply:
Yes. we needed it to see if same data collectors are not added.